airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -339,6 +378,14 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
 
     @staticmethod
     def instantiate_report_class(report: dict, config: Mapping[str, Any]) -> GoogleAnalyticsDataApiBaseStream:
-        return type(report["name"], (GoogleAnalyticsDataApiBaseStream,), {})(
-            config=dict(**config, metrics=report["metrics"], dimensions=report["dimensions"]), authenticator=config["authenticator"]
-        )
+        cohort_spec = report.get("cohortSpec")
+        pivots = report.get("pivots")
+        stream_config = {"metrics": report["metrics"], "dimensions": report["dimensions"], **config}
+        report_class_tuple = (GoogleAnalyticsDataApiBaseStream,)
+        if pivots:
+            stream_config["pivots"] = pivots
+            report_class_tuple = (PivotReport,)
+        if cohort_spec:
+            stream_config["cohort_spec"] = cohort_spec
+            report_class_tuple = (CohortReportMixin, *report_class_tuple)
+        return type(report["name"], report_class_tuple, {})(config=stream_config, authenticator=config["authenticator"])
```

Comment:
If both `pivots` and `cohort_spec` are present, I think object returned from this method is just going to have `CohortReportMixin`'s `request_body_json` method, which doesn't include the `"pivots"` key. Is that expected?

Reply:
method `request_body_json` in both classes relies on payload returned by invoking of the same method through `super()` thus it should include information both for `pivots` and `cohort_spec`

Reply:
Ah right, I see.